pulpcore/cli/common/generic.py

```diff
@@ -435,3 +476,100 @@ def label_show(entity_ctx: PulpEntityContext, key: str) -> None:
         label_group.add_command(subcmd)
 
     return label_group
+
+
+def repository_content_command(**kwargs: Any) -> click.Group:
+    """A factory that creates a repository content command group."""
+
+    content_contexts = kwargs.pop("contexts", {})
+    names = list(content_contexts.keys()) + ["all"]
+    content_contexts.update({"all": PulpContentContext})
+
+    @click.command("list")
+    @click.option("-t", "--type", "type", type=click.Choice(names), default=names[0])
+    @limit_option
+    @offset_option
+    @repository_option
+    @click.option("--base-version", type=int, callback=_version_callback, expose_value=False)
```

Comment:
I think this should be `--version`, `--repository-version`, or something similar. 

The term "base_version" is specific to repository version modification to indicate the version that is being used as a base to create the new repo version.